udev/10-azure-unmanaged-sriov.rules

```diff
@@ -0,0 +1,4 @@
+# Azure VMs with accelerated networking may have MANA, mlx4, or mlx5 SR-IOV devices which are transparently bonded to a synthetic
+# hv_netvsc device.  Mark these as unmanaged devices (NM_UNMANAGED=1 for NetworkManager, AZURE_UNMANAGED_SRIOV=1 for networkd, etc.).
+SUBSYSTEM=="net", ACTION!="remove", DRIVERS=="mana", ATTR{flags}=="0x?[89ABCDEF]??", ENV{NM_UNMANAGED}="1", ENV{AZURE_UNMANAGED_SRIOV}="1"
+SUBSYSTEM=="net", ACTION!="remove", DRIVERS=="hv_pci", DRIVERS=="mlx4_core|mlx5_core", ENV{NM_UNMANAGED}="1", ENV{AZURE_UNMANAGED_SRIOV}="1"
```

Comment:
Can you check on your end if this double `DRIVERS` match works? I tested on D2s_v5 and its not matching the mlx5 device at all:
```
$ udevadm info -q env -p /sys/class/net/enP41547s1
DEVPATH=/devices/LNXSYSTM:00/LNXSYBUS:00/ACPI0004:00/VMBUS:00/966a53df-a24b-44a0-94d3-6cddc1fa8987/pcia24b:00/a24b:00:02.0/net/enP41547s1
INTERFACE=enP41547s1
IFINDEX=3
SUBSYSTEM=net
USEC_INITIALIZED=22297092
ID_NET_DRIVER=mlx5_core
NVME_HOST_IFACE=none
ID_NET_NAMING_SCHEME=v255
ID_NET_NAME_MAC=enx000d3a4bc078
ID_OUI_FROM_DATABASE=Microsoft Corp.
ID_NET_NAME_PATH=enP41547p0s2
ID_NET_NAME_SLOT=enP41547s1
ID_BUS=pci
ID_VENDOR_ID=0x15b3
ID_MODEL_ID=0x1018
ID_PCI_CLASS_FROM_DATABASE=Network controller
ID_PCI_SUBCLASS_FROM_DATABASE=Ethernet controller
ID_VENDOR_FROM_DATABASE=Mellanox Technologies
ID_MODEL_FROM_DATABASE=MT27800 Family [ConnectX-5 Virtual Function]
ID_PATH=acpi-VMBUS:00-pci-a24b:00:02.0
ID_PATH_TAG=acpi-VMBUS_00-pci-a24b_00_02_0
ID_NET_LINK_FILE=/usr/lib/systemd/network/99-default.link
ID_NET_NAME=enP41547s1
SYSTEMD_ALIAS=/sys/subsystem/net/devices/enP41547s1
TAGS=:systemd:
CURRENT_TAGS=:systemd:
```

It works if I rewrite line 3 to:
```
SUBSYSTEM=="net", ACTION!="remove", DRIVERS=="mana|mlx4_core|mlx5_core", ATTR{flags}=="0x?[89ABCDEF]??", ENV{NM_UNMANAGED}="1", ENV{AZURE_UNMANAGED_SRIOV}="1"
```
```
$ udevadm info -q env -p /sys/class/net/enP41547s1
DEVPATH=/devices/LNXSYSTM:00/LNXSYBUS:00/ACPI0004:00/VMBUS:00/966a53df-a24b-44a0-94d3-6cddc1fa8987/pcia24b:00/a24b:00:02.0/net/enP41547s1
INTERFACE=enP41547s1
IFINDEX=3
SUBSYSTEM=net
USEC_INITIALIZED=22297092
NM_UNMANAGED=1
AZURE_UNMANAGED_SRIOV=1
ID_NET_DRIVER=mlx5_core
NVME_HOST_IFACE=none
ID_NET_NAMING_SCHEME=v255
ID_NET_NAME_MAC=enx000d3a4bc078
ID_OUI_FROM_DATABASE=Microsoft Corp.
ID_NET_NAME_PATH=enP41547p0s2
ID_NET_NAME_SLOT=enP41547s1
ID_BUS=pci
ID_VENDOR_ID=0x15b3
ID_MODEL_ID=0x1018
ID_PCI_CLASS_FROM_DATABASE=Network controller
ID_PCI_SUBCLASS_FROM_DATABASE=Ethernet controller
ID_VENDOR_FROM_DATABASE=Mellanox Technologies
ID_MODEL_FROM_DATABASE=MT27800 Family [ConnectX-5 Virtual Function]
ID_PATH=acpi-VMBUS:00-pci-a24b:00:02.0
ID_PATH_TAG=acpi-VMBUS_00-pci-a24b_00_02_0
ID_NET_LINK_FILE=/usr/lib/systemd/network/99-default.link
ID_NET_NAME=enP41547s1
SYSTEMD_ALIAS=/sys/subsystem/net/devices/enP41547s1
TAGS=:systemd:
CURRENT_TAGS=:systemd:
```

I'd also suggested adding:
```
ENV{ID_NET_MANAGED_BY}="unmanaged"
```
which is supported automatically in newer versions of networkd and networkmanager (https://github.com/systemd/systemd/pull/29782)